### ISSUE TITLE
tensorflow version corrosponding keras

### ISSUE THREAD

**Issue by jinxsfe** [googlecolab/colabtools#5212]
according the issue https://github.com/googlecolab/colabtools/issues/5099#issuecomment-2654355045..

I can install successfully for "!pip install -qq tensorflow[and-cuda]==2.15.0 tf-keras~=2.15.0 tensorrt-libs==8.6.1 --extra-index-url https://pypi.nvidia.com",

![Image](https://github.com/user-attachments/assets/2326a620-de7f-47d0-a876-a2a6983ff06a)
also tensorflow version and keras model seems worked, original pip install is,

![Image](https://github.com/user-attachments/assets/953c3e7a-105e-4786-9ba9-c741046bab27)

which got corresponding error 

![Image](https://github.com/user-attachments/assets/cce43004-47a0-485a-9be2-b4a8f4841017),

**Comment by jinxsfe** [comment]
if I run original method 

![Image](https://github.com/user-attachments/assets/bfdc7f13-2b02-40d8-b9b6-562bb2e0af4b), after restart the session, I skip that pip install, which can check version 

![Image](https://github.com/user-attachments/assets/68c6c728-f38d-45c3-aa56-1a68367267f9) version is 2.15, and during training 

![Image](https://github.com/user-attachments/assets/b118c82c-442c-4dfd-b2e4-6bb655337972)

![Image](https://github.com/user-attachments/assets/d703b555-834e-474d-865c-40cc5ef24604) the GPU are not utilized it

**Comment by metrizable** [comment]
@jinxsfe Thanks for filing the issue and thanks for using Colab.

Although it's hard to tell, it appears from your screenshots you're installing the CPU version of tensorflow, which will not take advantage of the GPU, rather than including the CUDA extras (`[and-cuda]`, from the [docs](https://www.tensorflow.org/install/pip)).

![Image](https://github.com/user-attachments/assets/ec04c7cf-1dc6-42c7-9d9d-f9acdeca7e98)

I would also recommend [reaching out](https://github.com/tensorflow/tensorflow/issues/new?template=tensorflow_issue_template.yaml) to the TensorFlow community for further assistance with your configuration.

**Comment by jinxsfe** [comment]
@metrizable  ，Dear advisor, thanks for your suggestion and advice, I retest the way as you mentioned before , the L4 and T4 gpu can work but A100 is failed, if I turn back A100 to L4, training is working for utilizes GPU. Is any different for A100 requirements? The newest way you suggest is !pip install -qq tensorflow[and-cuda]==2.15.0 tf-keras~=2.15.0 tensorrt-libs==8.6.1 --extra-index-url https://pypi.nvidia.com/", after doing previous commands you mentioned, I  restart session I run this commands without tensorflow and keras == 2.15, is working L4 and T4, thanks for your pain and advice.

![Image](https://github.com/user-attachments/assets/c2f1c1ad-3889-4e9e-9ea0-28b6178096fd)
